fiftyone/core/odm/document.py

```diff
@@ -382,22 +382,9 @@ def _get_repr_fields(self):
         # pylint: disable=no-member
         return self._fields_ordered
 
-    def to_dict(self, extended=False, no_dereference=False):
```

Comment:
It would in principle make sense to add a `no_dereference` flag to this base class, but the implementation would need to be improved to handle any `ReferenceField`.

The problem with the implementation here is that it contains details specific to saved views/runs, like how `serialize()` is called on lists and `to_dict()` is called on dict keys. If we can't have something generic here, I think we should only add this flag on the `DatasetDocument` class.

Reply:
Making it generic may be tricky or at least expensive because you'd have to traverse the entire document tree and check for `ReferenceField` types. I think it is fine to just hard code `no_dereference` in `DatasetDocument` for now, as it is the only usage of reference fields.

Reply:
yeah that sounds good.

By the way, I had been looking into `mongoengine.context_managers.no_dereference` for a generic way to handle this, but I couldn't get it to obey flag. It seems like the way we wrap the mongo document types or maybe how we access the data prevents it from working as expected. 
http://docs.mongoengine.org/guide/querying.html#turning-off-dereferencing